src/intersection/compute_intersection.cc

```diff
@@ -0,0 +1,1139 @@
+//     C.print();
+//     V.print();
+//     K.print();
+//     Det.print();
+//     cout <<endl;
+
+    unsigned int max_index = 0;
+    double maximum = fabs(Det[0]);
+    if(fabs((double)Det[1]) > maximum){
+        maximum = fabs(Det[1]);
+        max_index = 1;
+    }
+    if(fabs((double)Det[2]) > maximum){
+        maximum = fabs(Det[2]);
+        max_index = 2;
+    }
+//     DBGMSG("det max: %e\n", maximum);
+    //abscissa is parallel to triangle side
+    if(std::abs(maximum) <= std::sqrt(rounding_epsilon)) return false;
+
+    // map maximum index in {-UxV} to i,j of subdeterminants
+    //              i j
+    // max_index 0: 1 2
+    //           1: 2 0  (switch  due to sign change)
+    //           2: 0 1
+    unsigned int i = (max_index+1)%3,
+                 j = (max_index+2)%3;
+                 
+    double DetX = -K[i]*V[j] + K[j]*V[i];
+    double DetY = -K[i]*U[j] + K[j]*U[i];
+
+    double s = DetX/Det[max_index];   //parameter on abscissa
+    double t = DetY/Det[max_index];   //parameter on triangle side
+
+    // change sign according to side orientation
+    if(RefElement<2>::normal_orientation(side)) t=-t;
+
+//     DBGMSG("s = %f; t = %f\n",s,t);
+
+    // if IP is inside of triangle side
+    if(t >= -geometry_epsilon && t <= 1+geometry_epsilon){
+
+        IP.set_orientation(2);  // set orientation as a pathologic case ( > 1)
+        // possibly set abscissa vertex {0,1}
+        if( fabs(s) <= geometry_epsilon)       { s = 0; IP.set_topology_A(0,0);}
+        else if(fabs(1-s) <= geometry_epsilon) { s = 1; IP.set_topology_A(1,0);}
+        else                         {        IP.set_topology_A(0,1);}   // no vertex, line 0, dim = 1
+        
+        // possibly set triangle vertex {0,1,2}
+        if( fabs(t) <= geometry_epsilon)       { t = 0; IP.set_topology_B(RefElement<2>::interact(Interaction<0,1>(side))[RefElement<2>::normal_orientation(side)],0);}
+        else if(fabs(1-t) <= geometry_epsilon) { t = 1; IP.set_topology_B(RefElement<2>::interact(Interaction<0,1>(side))[1-RefElement<2>::normal_orientation(side)],0);}
+        else                         {        IP.set_topology_B(side,1);}   // no vertex, side side, dim = 1
+        
+        arma::vec2 local_abscissa({1-s, s});
+        arma::vec3 local_triangle({0,0,0});
+
+        // set local triangle barycentric coordinates according to nodes of the triangle side:
+        local_triangle[RefElement<2>::interact(Interaction<0,1>(side))[RefElement<2>::normal_orientation(side)]] = 1 - t;
+        local_triangle[RefElement<2>::interact(Interaction<0,1>(side))[1-RefElement<2>::normal_orientation(side)]] = t;
+//      local_abscissa.print();
+//      local_triangle.print();
+        
+        IP.set_coordinates(local_abscissa,local_triangle);
+        return true; // IP found
+    }
+    
+    return false;   // IP NOT found
+}
+
+
+bool ComputeIntersection<Simplex<1>, Simplex<2>>::compute(std::vector<IntersectionPointAux<1,2>> &IP12s, 
```

Comment:
I propose to distinguish:
- special case, when just 1 or 2 products are zero, this mean we have regular intersection but just on the boundary (edge, vertex). So in this case we comput as in the regular case and just mark topology info.
- degenerate case, when all 3 are zero, so the line and triangle are coplanar